battlepass/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+//      _______  ________  ________  ________   ______   _______   _______
+//    ╱╱       ╲╱        ╲╱        ╲╱        ╲_╱      ╲╲╱       ╲╲╱       ╲╲
+//   ╱╱      __╱         ╱         ╱         ╱        ╱╱        ╱╱        ╱╱
+//  ╱       ╱ ╱         ╱         ╱        _╱         ╱         ╱         ╱
+//  ╲________╱╲___╱____╱╲__╱__╱__╱╲________╱╲________╱╲___╱____╱╲________╱
+//
+// This file is part of GameDAO Protocol.
+// Copyright (C) 2018-2022 GameDAO AG.
+// SPDX-License-Identifier: Apache-2.0
+
+//! BATTLEPASS
+//! This pallet provides functionality to create, manage and participate in battlepasses.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_std::convert::TryInto;
+use sp_runtime::traits::Hash;
+use gamedao_traits::ControlTrait;
+use rmrk_traits::{primitives::PartId, Collection, Nft, ResourceInfoMin};
+
+pub mod types;
+pub use types::*;
+
+pub type String<T> = BoundedVec<u8, <T as Config>::StringLimit>;
+pub type Symbol<T> = BoundedVec<u8, <T as Config>::SymbolLimit>;
+pub type Resource<T> = BoundedVec<
+	ResourceInfoMin<
+		BoundedVec<u8, <T as Config>::StringLimit>,
+		BoundedVec<PartId, <T as Config>::PartsLimit>,
+	>,
+	<T as Config>::MaxResourcesOnMint,
+>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+  
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_rmrk_core::Config {
+		type Event: From<Event<Self>>
+			+ IsType<<Self as frame_system::Config>::Event>
+			+ Into<<Self as frame_system::Config>::Event>;
+
+		type Control: ControlTrait<Self::AccountId, Self::Hash>;
+
+		type Rmrk: Collection<String<Self>, Symbol<Self>, Self::AccountId>
+			+ Nft<Self::AccountId, String<Self>, Resource<Self>>;
+
+		/// The maximum length of a name, cid or metadata strings stored on-chain.
+		#[pallet::constant]
+		type StringLimit: Get<u32>;
+
+		/// The maximum length of a Collection symbol.
+		#[pallet::constant]
+		type SymbolLimit: Get<u32>;
+
+		/// The maximum number of parts each resource may have
+		#[pallet::constant]
+		type PartsLimit: Get<u32>;
+
+		type MaxResourcesOnMint: Get<u32>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// New BattlePass created
+		BattlepassCreated {
```

Comment:
I'd suggest using "Created", "Claimed", "Closed" instead. Like we have for Flow. Since the subject is always the same - Battlepass.

Reply:
There will be other entities in the pallet, like `Quests`, `Achievements`, etc which will have similar events.
Of course, we may put them in separate pallets, but they are related only to Battlepass, so I'm not sure if we should do so.
WDYT @vovacha , @vayesy , @2075 ?